app/test/check_tx_test.go

```diff
@@ -111,49 +110,41 @@ func TestCheckTx(t *testing.T) {
 			name:      "1,000 byte blob",
 			checkType: abci.CheckTxType_New,
 			getTx: func() []byte {
-				tx := blobfactory.RandBlobTxsWithAccounts(encCfg.TxConfig.TxEncoder(), tmrand.NewRand(), kr, nil, 1_000, 1, false, testutil.ChainID, accs[4:5])[0]
-				return tx
+				return blobfactory.BlobTxWithSize(t, encCfg.TxConfig.TxEncoder(), kr, testutil.ChainID, accs[4], 1_000)
 			},
 			expectedABCICode: abci.CodeTypeOK,
 		},
 		{
 			name:      "10,000 byte blob",
 			checkType: abci.CheckTxType_New,
 			getTx: func() []byte {
-				tx := blobfactory.RandBlobTxsWithAccounts(encCfg.TxConfig.TxEncoder(), tmrand.NewRand(), kr, nil, 10_000, 1, false, testutil.ChainID, accs[5:6])[0]
-				return tx
+				return blobfactory.BlobTxWithSize(t, encCfg.TxConfig.TxEncoder(), kr, testutil.ChainID, accs[5], 10_000)
 			},
 			expectedABCICode: abci.CodeTypeOK,
 		},
 		{
 			name:      "100,000 byte blob",
 			checkType: abci.CheckTxType_New,
 			getTx: func() []byte {
-				tx := blobfactory.RandBlobTxsWithAccounts(encCfg.TxConfig.TxEncoder(), tmrand.NewRand(), kr, nil, 100_000, 1, false, testutil.ChainID, accs[6:7])[0]
-				return tx
+				return blobfactory.BlobTxWithSize(t, encCfg.TxConfig.TxEncoder(), kr, testutil.ChainID, accs[6], 100_000)
 			},
 			expectedABCICode: abci.CodeTypeOK,
 		},
 		{
 			name:      "1,000,000 byte blob",
 			checkType: abci.CheckTxType_New,
 			getTx: func() []byte {
-				tx := blobfactory.RandBlobTxsWithAccounts(encCfg.TxConfig.TxEncoder(), tmrand.NewRand(), kr, nil, 1_000_000, 1, false, testutil.ChainID, accs[7:8])[0]
-				return tx
+				return blobfactory.BlobTxWithSize(t, encCfg.TxConfig.TxEncoder(), kr, testutil.ChainID, accs[7], 1_000_000)
 			},
 			expectedABCICode: abci.CodeTypeOK,
```

Comment:
[comment] 

how are these tests passing without account numbers being added? shouldn't the signature be failing? also it appears we have two `RandBlobTxsWithAccounts`, one in `util` that uses the app to query the account number and sequence directly from the testapp, and then another in blobfactor where we don't...

Reply:
I'm not sure what you mean by account numbers being added? This test uses `accs[7]`, other test cases use unique accounts.

Reply:
Refactored to use `RandBlobTxsWithAccounts` in all test cases except the last one b/c the last one would fail to create a blob tx for a 10,000,000 byte blob if it used `RandBlobTxsWithAccounts`.

 https://github.com/celestiaorg/celestia-app/pull/2202/commits/d8faa9760ca88b3ca9c928c9bc385bdf5ff04c91

Reply:
similar to the sequence or chain id, the account number of a given account is added to the tx sign bytes when we sign the tx but its not actually included in the transaction, therefore we have to look it up before we sign a tx.

since we're not looking up sequence or account number (passing nil for the conn arg), I would have expected the signature of the tx to fail...

the one in util can query directly using the testapp (doesn't use grpc). I'll take another look at this test and see if there's something wrong

Reply:
Resolving this b/c no longer applicable.